runtime/lua/vim/lsp.lua

```diff
@@ -78,7 +78,7 @@ end
 
 --- Returns the buffer number for the given {bufnr}.
 ---
----@param bufnr (integer|nil) Buffer number to resolve. Defaults to current buffer
+---@param bufnr (integer?) Buffer number to resolve. Defaults to current buffer
```

Comment:
```suggestion
---@param bufnr? integer Buffer number to resolve. Defaults to current buffer
```

The optional marker goes on the argument name, not type, and parens are no longer needed. Please change that everywhere.

Reply:
Except return type; there it's correct, I think.

Reply:
It can go on both. When on the type it indicates the value may be nil but should be provided, and when on the arg name it indicates the argument is completely optional.

In practice, LuaLS won't ever be able to check this, but there is a semantic difference. In Lua you can detect the number of call arguments with `select('#', ...)` for vararg functions which will give you 2 for `foo(nil, nil)`.

Reply:
I believe it makes a difference for the dreaded "missing field" diagnostic, which somewhat triggered this push...

(Apart from that, my argument is mostly from consistency and "neatness", I have to admit.)

Reply:
I've updated the PR to have the `?` in the appropriate place.